js/background_scripts/background.js

```diff
@@ -74,43 +74,79 @@ chrome.runtime.onMessage.addListener(
 //Setup Icon Paths
 var notCompatibleIconPath = chrome.runtime.getURL("images/iconNotCompatible.png");
 var CompatibleIconPath = chrome.runtime.getURL("images/icon.png");
+var allTabs = {}
 
-//Check if the newly Selected Tab is playable
+//Check if the newly Selected Tab is compatible
 chrome.tabs.onActivated.addListener(function(info){
-    chrome.tabs.getSelected(null,function(tab){
-        var tabURL = tab.url;
+    chrome.tabs.query({active: true,lastFocusedWindow: true}, function (tab) {
+        var tabURL = tab[0].url;
+        var tabID = tab[0].id;
+        
+        //Check if Tab has the required infos
+        if(allTabs[tabID] == null){
+            allTabs[tabID] = {needCheck: true, isCompatible: false};
+        }
 
-        for (var i = 0; i < allModules.length; i++) {
-            var module = allModules[i];
-            if (module.canHandleUrl(tabURL) && !isContextMenuCreated(tabURL)){
+        //Check if Tab needs to be Checked
+        if(!allTabs[tabID].needCheck){
+            if(allTabs[tabID].isCompatible)
                 chrome.browserAction.setIcon({path: CompatibleIconPath});
-                break;
-            }
-            //Reduce Flickering of Icon
-            if(i == allModules.length-1)
+            else
                 chrome.browserAction.setIcon({path: notCompatibleIconPath});
+            return;
         }
+
+        //Check if Tab is Compatible
+        checkIfTabIsCompatible(tabURL,tabID);
```

Comment:
Why `checkIfTabIsCopatible` is run at the end?
It won't set the icon during this run when the tab `needChec = true`

I think this should rather look like this:
```
        //Check if Tab needs to be Checked
        if(allTabs[tabID].needCheck) {
                //Check if Tab is Compatible
                checkIfTabIsCompatible(tabURL,tabID);
        }
        if(allTabs[tabID].isCompatible) {
           chrome.browserAction.setIcon({path: CompatibleIconPath});
        } else {
           chrome.browserAction.setIcon({path: notCompatibleIconPath});
        }
```

Reply:
checkIfTabIsCompatible is run at the end, because i added the function in the end and didn´t really think about where to put it.

The function itself changes the icon after the check.

I really like the way you did it, it looks a lot cleaner than mine.
I also really like all the help and suggestions you have ♥ Thank you for that.